Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,120 @@ static void Main(string[] args)
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
+        public static int FindDigit(string equation)
+        {
+            int index = equation.IndexOf('*');
+            String a = equation.Substring(0,index);
+
+            int index1 = equation.IndexOf('=');
+            String b = equation.Substring(index+1,index1-index-1);
+
+            String c = equation.Substring(index1+1);
+
+            if(a.Contains('?'))  {
+                float num2 = float.Parse(b);
+                float res = float.Parse(c);
+
+                float num1 = res/num2;
+
+                String result = num1.ToString();
+
+                char[] a1 = a.ToCharArray();
+                char[] res1 = result.ToCharArray();
+
+                if(a1.Length == res1.Length)  {
+                    int i = 0, num = -1;
+                    while(i<a1.Length)  {
+                        if(a1[i] == '?')  {
+                            num = int.Parse(res1[i].ToString());
+                            i++;
+                        }
+                        else if(a1[i] == res1[i])  {
+                            i++;
+                        }
+                        else
+                            break;
+                    }
+                    if(i == a1.Length)
+                        return num;
+                    else
+                        return -1;
+                }
+                else
+                    return -1;
+            }
+
+            else if(b.Contains('?'))  {
+                float num1 = float.Parse(a);
+                float res = float.Parse(c);
+
+                float num2 = res/num1;
+
+                String result = num2.ToString();
+
+                char[] b1 = b.ToCharArray();
+                char[] res1 = result.ToCharArray();
+
+                if(b1.Length == res1.Length)  {
+                    int i = 0, num = -1;
+                    while(i<b1.Length)  {
+                        if(b1[i] == '?')  {
+                            num = int.Parse(res1[i].ToString());
+                            i++;
+                        }
+                        else if(b1[i] == res1[i])  {
+                            i++;
+                        }
+                        else
+                            break;
+                    }
+                    if(i == b1.Length)
+                        return num;
+                    else
+                        return -1;
+                }
+                else
+                    return -1;
+            }
+
+            else  {
+                float num1 = float.Parse(a);
+                float num2 = float.Parse(b);
+
+                float res = num1 * num2;
+
+                String result = res.ToString();
+
+                char[] c1 = c.ToCharArray();

```

Comment:
Please work on the naming conventions of the variables. Names like `c1` and `res1` don't make any sense for me unless I review the code in detail. A variable name should indicate the usage of that variable.